src/app/modules/querybuilder/components/querybuilder-editor/edit/edit-criterion/edit-criterion.component.ts

```diff
@@ -62,13 +72,72 @@ export class EditCriterionComponent implements OnInit, AfterViewChecked {
     }
 
     this.showGroups = this.query.groups.length > 1
+
+    if (!this.featureService.mockLoadnSave()) {
+      this.loadUIProfile()
+    }
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptionCritProfile?.unsubscribe()
   }
 
   ngAfterViewChecked(): void {
     this.actionDisabled = this.isActionDisabled()
     this.changeDetector.detectChanges()
   }
 
+  loadUIProfile(): void {
+    this.subscriptionCritProfile?.unsubscribe()
+    const version = this.criterion.termCodes[0].version
+      ? '&version=' + this.criterion.termCodes[0].version
+      : ''
+    const param =
+      'code=' +
+      this.criterion.termCodes[0].code +
+      '&system=' +
+      this.criterion.termCodes[0].system +
+      version
+    this.subscriptionCritProfile = this.backend
+      .getTerminologyProfile(param)
+      .subscribe((profile) => {
+        if (profile.timeRestrictionAllowed && !this.criterion.timeRestriction) {
+          this.criterion.timeRestriction = { tvpe: TimeRestrictionType.BETWEEN }
```

Comment:
@thkoehler11  tvpe  looks like a spelling mistake - please double check - should be type ?